src/api.ts

```diff
@@ -301,13 +293,15 @@ async function getRemoteBranches(): Promise< Map< string, string > > {
 		);
 
 		repo.free();
-		// gc the repo
-		try {
-			await promisify( exec )( 'git gc', {
-				cwd: calypsoDir,
-			} );
-		} catch ( err ) {
-			l.error( { err }, 'git gc failed' );
+		// gc the repo if no builds are running
+		if ( pendingHashes.size === 0 ) {
+			try {
+				await promisify( exec )( 'git gc', {
```

Comment:
I like to _be aggressive_.

```suggestion
				await promisify( exec )( 'git gc --prune=all --aggressive', {
```

Reply:
Are you sure? https://git-scm.com/docs/git-gc#_aggressive

> Not using this at all is the right trade-off for most users and their repositories.

Reply:
and we already do the prune as part of the fetch